numpyro/optim.py

```diff
@@ -53,8 +53,11 @@ def _wrapper(x):
 class _NumPyroOptim(object):
     def __init__(self, optim_fn: Callable, *args, **kwargs) -> None:
         self.init_fn: Callable[[_Params], _IterOptState]
-        self.update_fn: Callable[[ArrayLike, _Params, _OptState], _OptState]
+        self.update_fn: Union[Callable[[ArrayLike, _Params, _OptState], _OptState], Callable[[ArrayLike, _Params, _OptState, ArrayLike], _OptState]]
         self.get_params_fn: Callable[[_OptState], _Params]
+        self.update_with_value: bool = False
+        if "update_with_value" in kwargs:
+            self.update_with_value = kwargs.pop("update_with_value")
```

Comment:
you can simply do `self.update_with_value = kwargs.pop("update_with_value", False)`